mPower2/mPower2/Custom Views/ProgressCircleView.swift

```diff
@@ -0,0 +1,141 @@
+// CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY,
+// OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+// OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+//
+
+import UIKit
+
+class ProgressCircleView: UIView {
+    
+    private let kVerticalPad: CGFloat = 12.0
+
+    let progressShape = CAShapeLayer()
+    let backgroundShape = CAShapeLayer()
+    
+    public var progress: Double = 30.0 {
+        didSet {
+            updateProgress()
+        }
+    }
+    lazy var dayLabel: UILabel = {
+        let label = UILabel()
+        label.translatesAutoresizingMaskIntoConstraints = false
+        label.font = UIFont(name: "HelveticaNeue-Bold", size: 12.0)
+        label.textColor = UIColor.royal700
+        label.textAlignment = .center
+        label.text = "Day" // TODO: jbruhin 5-1-18 localize
+        label.sizeToFit()
+        addSubview(label)
+        label.rsd_alignCenterHorizontal(padding: 0.0)
+        label.rsd_alignToSuperview([.top], padding: kVerticalPad)
+        return UILabel()
+    }()
+    lazy var dayCountLabel: UILabel = {
+        let label = UILabel()
+        label.translatesAutoresizingMaskIntoConstraints = false
+        label.font = UIFont(name: "HelveticaNeue-Thin", size: 30.0)
+        label.textColor = UIColor.royal700
+        label.textAlignment = .center
+        addSubview(label)
+        label.rsd_alignCenterVertical(padding: 5.0)
+        label.rsd_alignToSuperview([.leading, .trailing, .bottom], padding: 0.0)
+        return label
+    }()
+    lazy var imageView: UIImageView = {
+        let imageView = UIImageView()
+        imageView.translatesAutoresizingMaskIntoConstraints = false
+        addSubview(imageView)
+        imageView.rsd_alignAllToSuperview(padding: 10.0)
+        return UIImageView()
+    }()
+
+
+    public override init(frame: CGRect) {
+        super.init(frame: frame)
+        commonInit()
+    }
+    
+    public required init?(coder aDecoder: NSCoder) {
+        super.init(coder: aDecoder)
+        commonInit()
+    }
+    
+    fileprivate func commonInit() {
+        layer.addSublayer(backgroundShape)
+        layer.addSublayer(progressShape)
+        updateProgress()
+    }
+
+    override func layoutSubviews() {
+        super.layoutSubviews()
+        progressShape.frame = bounds
+        backgroundShape.frame = bounds
+    }
+    
+    func updateProgress() {
+        
+        let shortestSide: CGFloat = min(self.frame.size.width, self.frame.size.height)
+        let strokeWidth: CGFloat = 4.0
+        let frame = CGRect(x: strokeWidth/2, y: strokeWidth/2, width: shortestSide - strokeWidth, height: shortestSide - strokeWidth)
+        
+        backgroundShape.frame = frame
+        backgroundShape.position = center
+        backgroundShape.path = UIBezierPath(ovalIn: frame).cgPath
+        backgroundShape.strokeColor = UIColor.royal200.cgColor
```

Comment:
This is basically the same thing as the `RSDCountdownDial` and for consistency, should use those colors.
`UIColor.rsd_dialRingBackground`

Which, by the by, is **not** purple in the Figma docs. It's gray.

Reply:
done